### PR TITLE
[no squash] Rendering and some other fixes

### PR DESCRIPTION
**Pull request by sfan5**
## To do

This PR is Ready for Review.

## How to test

1. check that colored text still works
2. check that transparent materials are still sorted correctly

### CODE REVIEW

**Comment by SmallJoker** [comment]
Just to make sure - 29735fc is not related to #14999, right?

**Comment by sfan5** [comment]
Nope.

File: irr/src/CNullDriver.h [hunk 1]
```diff
@@ -570,6 +566,11 @@ class CNullDriver : public IVideoDriver, public IGPUProgrammingServices
 	// prints renderer version
 	void printVersion();
 
+	inline void accountHWBufferUpload(u32 size) {
+		FrameStats.HWBuffersUploaded++;
```
Comment (Desour) [inline irr/src/CNullDriver.h:570]:
```suggestion
	inline void accountHWBufferUpload(u32 size)
	{
		FrameStats.HWBuffersUploaded++;
```

**Review by Desour — APPROVED** [review]
Looks good.

**Pull request merged**